tests/foreman/ui_airgun/test_repository.py

```diff
@@ -482,3 +493,279 @@ def test_positive_resynchronize_puppet_repo(session, module_prod):
+                'repo_content.upstream_username': upstream_username,
+                'repo_content.upstream_password': upstream_password,
+            }
+        )
+        assert session.repository.search(module_prod.name, repo_name)[0]['Name'] == repo_name
+        repo_values = session.repository.read(module_prod.name, repo_name)
+        assert repo_values['repo_content']['upstream_authorization'] == '{0} / {1}'.format(
+            upstream_username, hidden_password)
+        session.repository.update(
+            module_prod.name,
+            repo_name,
+            {
+                'repo_content.upstream_authorization': dict(
+                    username=new_upstream_username,
+                    password=new_upstream_password
+                ),
+            }
+        )
+        repo_values = session.repository.read(module_prod.name, repo_name)
+        assert repo_values['repo_content']['upstream_authorization'] == '{0} / {1}'.format(
+            new_upstream_username, hidden_password)
+        session.repository.update(
+            module_prod.name,
+            repo_name,
+            {
+                'repo_content.upstream_authorization': dict(username=''),
+            }
+        )
+        repo_values = session.repository.read(module_prod.name, repo_name)
+        assert not repo_values['repo_content']['upstream_authorization']
+
+
+@tier2
+@upgrade
+def test_positive_end_to_end_custom_ostree_crud(session, module_prod):
+    """Perform end to end testing for custom ostree repository
+
+    :id: 603372aa-60de-44a8-b6c9-3f84c3bbdf05
+
+    :expectedresults: All expected CRUD actions finished successfully
+
+    :CaseLevel: Integration
+
+    :CaseImportance: High
+
+    :BZ: 1467722
+    """
+    repo_name = gen_string('alpha')
+    new_repo_name = gen_string('alphanumeric')
+    with session:
+        session.repository.create(
+            module_prod.name,
+            {
+                'name': repo_name,
+                'repo_type': REPO_TYPE['ostree'],
+                'repo_content.upstream_url': FEDORA22_OSTREE_REPO,
+            }
+        )
+        assert session.repository.search(module_prod.name, repo_name)[0]['Name'] == repo_name
+        session.repository.update(
+            module_prod.name,
+            repo_name,
+            {
+                'name': new_repo_name,
+                'repo_content.upstream_url': FEDORA23_OSTREE_REPO,
+            }
+        )
+        assert not session.repository.search(module_prod.name, repo_name)
+        repo_values = session.repository.read(module_prod.name, new_repo_name)
+        assert repo_values['name'] == new_repo_name
+        assert repo_values['repo_content']['upstream_url'] == FEDORA23_OSTREE_REPO
+        session.repository.delete(module_prod.name, new_repo_name)
+        # a task is created for delete operation and we are not redirected to that task page.
+        # this operation can take some time.
+        for _ in range(10):
```

Comment:
same comment here about task and `wait_for`

Reply:
prefer to implement it at airgun side instead